src/main/java/org/dependencytrack/tasks/repositories/RepositoryMetaAnalyzerTask.java

```diff
@@ -155,21 +155,24 @@ private void analyze(final QueryManager qm, final Component component, final IMe
                 ComponentAnalysisCache cac = cacByHost.get(repository.getUrl());
                 MetaModel model = new MetaModel(component);
                 if (cac != null && isCacheCurrent(cac, component.getPurl().toString())) {
-                    LOGGER.debug("Building repository Metamodel from cache for "+purl);
+                    LOGGER.debug("Building repository Metamodel from cache for " + purl);
                     model.setLatestVersion(StringUtils.trimToNull(cac.getResult().getString(LATEST_VERSION)));
                     model.setPublishedTimestamp(Date.from(Instant.ofEpochMilli(cac.getResult().getJsonNumber(PUBLISHED_TIMESTAMP).longValue())));
                 } else {
                     LOGGER.debug("Analyzing component: " + component.getUuid() + " using repository: "
                             + repository.getIdentifier() + " (" + repository.getType() + ")");
 
-                    if (repository.isInternal()) {
+                    if (Boolean.TRUE.equals(repository.isAuthenticationRequired())) {
```

Comment:
For existing installations, keep in mind that `isAuthenticationRequired` will be `null`, but `isInternal` might be `true` (again, unless we add a migration that flips `isAuthenticationRequired` to `true` for all repos that are internal and have credentials configured).

```suggestion
                    if (Boolean.TRUE.equals(repository.isInternal()) || Boolean.TRUE.equals(repository.isAuthenticationRequired())) {
```